<h1> $\beta_i = 1$ plasmas </h1>

<h2> Summary </h2>
- We made linear runs to choose the driving amplitudes that should pump comparable energy into slow modes and Alfven modes.
- We verified that the chosen ion and electron collisionality work well to produce Alfven dispersion relation that agree reasonably well with dispersion relation in the hot kinetic plasmas.
- We did nonlinear runs (still running) on a smaller grid ($32\times 32\times 32$ in physical space) on a single Haswell node (32 processors) on Nersc/Cori.

<h2> linear simulation</h2>
In hot kinetic plasma:
- Alfven mode $\overline{\omega}_0=1.005 -0.002 i$
- slow mode $\overline{\omega}_0=1.25-0.88i$

Figures below show the Alfven and slow modes in linear gyrokinetic simulations. 
- Alfven mode is well excited and decays very slowly
- $u_\parallel$ antenna excite both Alfven and slow modes since the two modes have very similar frequencies. 
- In case of $u_\parallel$ antenna driving, $MVA\sim 0.1$ => $E_{slow} \sim 10 E_{alfven}$

<table border="0">
    <tr>
        <td style="padding:5px">
            <img src='../figures/lin_bi_1_alfven_Ebperp_1.png' width="400">
        </td>
        <td style="padding:5px">
            <img src='../figures/lin_bi_1_alfven_mva_1.png' width="400">         
        </td>
    </tr>
</table>

<b>Figure 1</b> $E_{B_\perp}$ and $MVA$ in a linear run with $A_\parallel$ antenna exciting Alfven waves.


<table border="0">
    <tr>
        <td style="padding:5px">
            <img src='../figures/lin_bi_1_slow_Ebpar_1.png' width="400">
        </td>
        <td style="padding:5px">
            <img src='../figures/lin_bi_1_slow_mva_1.png' width="400">         
        </td>
    </tr>
</table>

<b>Figure 2</b> $E_{B_\parallel}$ and $MVA$ in a linear run with $u_\parallel$ antenna exciting slow waves.

<h2> confirm collision rate </h2>

- For the linear run, we used $\mu_i=0.1$ and $\mu_e=0.5$. 
- Below we check that the AstroGK linear dispersion relation is close to the hot kinetic plasma Alfven solution.
- We skip the comparison for slow waves because it is too heavily damped to obtain dispersion relation in AstroGK linear simulation.

In [1]:
import pandas as pd

We fit $E_{B_{\perp}} (t)$ by exponential model and obtain dispersion relation of the Alfven modes in the AstroGK linear simulations.

In [2]:
df = pd.DataFrame()
kperp = [0.2, 1.0, 2.0, 4.0, 8.0]
hot_wr = [1.00508064, 1.13655115, 1.57770556 ,2.8990671, 5.71804829]
hot_wi = [-0.00185854, -0.0327562, -0.06670535, -0.21006724, -0.81787136]
agk_wr = [1.01, 1.142, 1.583, 2.911, 5.72]
agk_wi = [-0.0023, -0.036, -0.075, -0.23, -0.86]
df[r'$k_\perp$'] = kperp
df[r'hot $\overline{\omega}_r$'] = hot_wr
df[r'hot $\overline{\omega}_i$'] = hot_wi
df[r'agk $\overline{\omega}_r$'] = agk_wr
df[r'agk $\overline{\omega}_i$'] = agk_wi
df[r'hot $\overline{\omega}_r$'] = df[r'hot $\overline{\omega}_r$'].map('{:,.2f}'.format)
df[r'hot $\overline{\omega}_i$'] = df[r'hot $\overline{\omega}_i$'].map('{:,.2g}'.format)
df[r'agk $\overline{\omega}_r$'] = df[r'agk $\overline{\omega}_r$'].map('{:,.2f}'.format)
df[r'agk $\overline{\omega}_i$'] = df[r'agk $\overline{\omega}_i$'].map('{:,.2g}'.format)

In [3]:
df

$k_\perp$ hot $\overline{\omega}_r$ hot $\overline{\omega}_i$  \
0        0.2                      1.01                   -0.0019   
1        1.0                      1.14                    -0.033   
2        2.0                      1.58                    -0.067   
3        4.0                      2.90                     -0.21   
4        8.0                      5.72                     -0.82   

  agk $\overline{\omega}_r$ agk $\overline{\omega}_i$  
0                      1.01                   -0.0023  
1                      1.14                    -0.036  
2                      1.58                    -0.075  
3                      2.91                     -0.23  
4                      5.72                     -0.86

<b>Table 1.</b> We compare the Alfven wave freqeuency in AstroGK linear simulations with that in hot kinetic plasma. 

- The real part agrees fairly well, the imaginary part has ~$5-15\%$ difference.

<h2> Driving amplitude </h2>

The Alfven waves are driven by the $A_0$ antenna at $\overline{\omega}_0=0.9$
- $E_{B_\parallel}=3.95\times10^{-4}A_0^2$ and $E_{B_\perp}=0.0584A_0^2$

The slow waves are driven by the $u_0$ antenna at $\overline{\omega}_0=1.1$
- $E_{B_\parallel}=0.35 u_0^2$ and $E_{B_\perp}=0.0363 u_0^2$

Hence we determine that to achieve equal energy in both modes, we need driving amplitude:
- $A_0 = 30$ (for strong turbulence)
- $u_0 = 12.9$

<table border="0">
    <tr>
        <td style="padding:5px">
            <img src='../figures/lin_bi_1_mix_Ebperp_1.png' width="400">
        </td>
        <td style="padding:5px">
            <img src='../figures/lin_bi_1_mix_mva_1.png' width="400">         
        </td>
    </tr>
</table>

<b>Figure 3</b> $E_{B_\perp}$ and $MVA$ in a linear run with both Alfven and slow mode.

<h2> Nonlinear simulations </h2>

We make nonliear runs on a smaller grid ($32\times 32\times 32$ in physical space) on a single Haswell node (32 processors) on Nersc/Cori. 
- The spectrum have not saturated yet. 
- So far, we couldn't see the difference between the difference between two simulations.

<table border="0">
    <tr>
        <td style="padding:5px">
            <video controls src="../videos/nl_bi_1_mix_mva_1.mp4", width="400"/>
          </td>
        <td style="padding:5px">
            <video controls src="../videos/nl_bi_1_mix_E_1.mp4", width="400"/>
         </td>
    </tr>
</table>

<b>Video 1</b> MVA and energy spectrum in a nonlinear run in which both Alfven and slow modes. The Alfven mode is driven by $A_0$ antenna with frequency $\overline{\omega}_0=0.9$ and decorrelation rate $DR=0.8$, and antenna amplitude $A_0=30$. The slow mode is driven by $A_0$ antenna with frequency $\overline{\omega}_0=1.1$ and decorrelation rate $DR=0$ and antenna amplitude $u_0=19.2$

<table border="0">
    <tr>
        <td style="padding:5px">
            <video controls src="../videos/nl_bi_1_alfven_mva_1.mp4", width="400"/>
          </td>
        <td style="padding:5px">
            <video controls src="../videos/nl_bi_1_alfven_E_1.mp4", width="400"/>
         </td>
    </tr>
</table>

<b>Video 2</b> MVA and energy spectrum in a nonlinear run with only Alfven mode.  The Alfven mode is driven by $A_0$ antenna with frequency $\overline{\omega}_0=0.9$ and decorrelation rate $DR=0.8$. The antenna amplitude is $A_0=30$.

<br><br>
<hr>
<br><br>
<h1> $\beta_i = 0.01$ plasmas </h1>


<h2> Summary </h2>
- We made a long expansion run
- Both electron and ion spectrum look bottlenecked. We look into adjusting hypercollisionality parameters.

<h2> After expansion</h2>
- double $k_x$ and $k_y$
- continue run on 512 processors

Run speed:
- step size = $4.88\times 10^{-6}$
- speed = 555 min / 18000 steps = 0.031 min/step

<h3> spectrum evolution: first long expanded run</h3>
<table border="0">
    <tr>
        <td style="padding:5px">
            <video controls src="../videos/nl_mix_large_mva_2.mp4", width="400"/>
          </td>
        <td style="padding:5px">
            <video controls src="../videos/nl_mix_large_E_2.mp4", width="400"/>
         </td>
    </tr>
</table>
<b>Video 3</b> MVA and energy spectrum in a nonlinear run with both Alfven and slow modes. 

<table border="0">
    <tr>
        <td style="padding:5px">
            <video controls src="../videos/nl_alfven_large_mva_2.mp4", width="400"/>
          </td>
        <td style="padding:5px">
            <video controls src="../videos/nl_alfven_large_E_2.mp4", width="400"/>
         </td>
    </tr>
</table>
<b>Video 4</b> MVA and energy spectrum in a nonlinear run with only Alfven and mode. 

<h2> Collision parameters </h2>

We look for $2\pi\gamma_{tot}\sim \omega_{NL}$. Below we show diagnostics of collisionality and hypercollisionality in a few runs restarting from the mixed mode run shown in Video 3. 
- short horizontal dashed lines in figures below reference $1/2\pi$

<table border="0">
    <tr>
        <td style="padding:2px">
            <img src='../figures/mix_large_4a_elec.png' width="300">
        </td>
        <td style="padding:2px">
            <img src='../figures/mix_large_4a_ion.png' width="300">         
        </td>
        <td style="padding:2px">
            <video controls src="../videos/nl_mix_large_E_2.mp4", width="300"/>
         </td>
    </tr>
</table>

<b>Figure 4</b> Electron `max_nuh`$=10$ and ion `nu_h`=1.

<table border="0">
    <tr>
        <td style="padding:2px">
            <img src='../figures/mix_large_4b_elec.png' width="300">
        </td>
        <td style="padding:2px">
            <img src='../figures/mix_large_4b_ion.png' width="300">         
        </td>
        <td style="padding:2px">
            <video controls src="../videos/nl_mix_large_E_4b.mp4", width="300"/>
         </td>        
    </tr>
</table>
<b>Figure 5</b> Electron `max_nuh`$=10^3$ and ion `nu_h`=1.

<table border="0">
    <tr>
        <td style="padding:2px">
            <img src='../figures/mix_large_4c_elec.png' width="300">
        </td>
        <td style="padding:2px">
            <img src='../figures/mix_large_4c_ion.png' width="300">         
        </td>
        <td style="padding:2px">
            <video controls src="../videos/nl_mix_large_E_4c.mp4", width="300"/>
        </td>        
    </tr>
</table>
<b>Figure 6</b> Electron `max_nuh`$=10^4$ and ion `nu_h`=1.

<table border="0">
    <tr>
        <td style="padding:2px">
            <img src='../figures/mix_large_4d_elec.png' width="300">
        </td>
        <td style="padding:2px">
            <img src='../figures/mix_large_4d_ion.png' width="300">         
        </td>
        <td style="padding:2px">
            <video controls src="../videos/nl_mix_large_E_4d.mp4", width="300"/>
        </td>        
    </tr>
</table>
<b>Figure 7</b> Electron `max_nuh`$=10^4$ and ion `nu_h`=10.

<table border="0">
    <tr>
        <td style="padding:2px">
            <img src='../figures/mix_large_4e_elec.png' width="300">
        </td>
        <td style="padding:2px">
            <img src='../figures/mix_large_4e_ion.png' width="300">         
        </td>      
    </tr>
</table>
<b>Figure 8</b> Electron `max_nuh`$=5\times10^3$ and ion `nu_h`=1.

It seems that Electron `max_nuh`$\sim 5000$ and ion `nu_h`=1 maybe better. (waiting for simulation to finish)